core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -529,6 +557,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       logError(t.getMessage, t))(ThreadUtils.sameThread)
   }
 
+  /**
+   * Called by subclasses when notified of a decommissioning worker.
+   */
+  private[spark] def decommissionExecutor(executorId: String): Unit = {
+    // Only log the failure since we don't care about the result.
```

Comment:
Can't this be a `send` in that case (which is cheaper)? Failures should already be logged by the RPC layer in that case.

Reply:
I think this was previously to avoid the case where the executor marked its self as decomissioned and stopped scheduling jobs before the driver knew to stop trying to schedule jobs on it, but we've changed that behaviour so it is no longer needed to do the expensive request.